### ISSUE TITLE
Develop a version of WebSurferAgent that uses a proper headless browser (perhaps with multimodal LLM support)

### ISSUE THREAD

**Issue by afourney** [microsoft/autogen#1481]
Possible roadmap:

- Investigate (Headless Chrome)[https://developer.chrome.com/blog/headless-chrome/] or equivalent.
- Add a HeadlessChromeBrowser to complement SimpleTextBrowser in https://github.com/microsoft/autogen/blob/main/autogen/browser_utils.py
- Update WebSurferAgent to accept a web_broswer instance rather than a web_browser_config, and pass in either a SimpleTextBrowser or HeadlessChromeBrowser as appropriate

Additional thoughts: We should try to take full advantage to having a browser under our control. Don't just dump the dom to HTML for BeautifulSoup to parse (like what Langchain does). Rather use javascript running privileged in the page context to query the document, extract text, interact with links, etc.

**Comment by vijaykramesh** [comment]
Started an initial pass at this, right now it doesn't do much more than the SimpleTextBrowser but since it runs in a headless chrome it can take advantage of client side interactions.  It might also be worth looking into something like https://github.com/lucgagan/auto-playwright to allow the HeadlessChromeBrowser agent to intelligently do selenium interactions.

**Comment by gee842** [comment]
I have opened this issue which might be related as it tackles a similar problem: #1538 

However, one enhancement that would be extremely beneficial would be shortening the context length by selectively remembering things and increasing the useful-information-density produced by the web browsing agent.

**Comment by INF800** [comment]
Hi @afourney @vijaykramesh, I want to try this feature but I see that this issue is still pending.  

If you guys don't mind I will try to complete this PR while experimenting with the headless browser.

You may see the merges from latest work in main and vijaykramesh's branch here: https://github.com/INF800/autogen/tree/feat/headless_browser

**Comment by signalprime** [comment]
Hi friends, if you're in a hurry to try it out, here is the PR https://github.com/microsoft/autogen/pull/1733 and on Friday I'll have time to finish the PR feedback.  To my surprise, agent navigation such as clicking on links continued to work even with the graphical browser.  Edge, chrome, and Firefox are supported.  It was a smaller building block in a larger pipeline I'd been working on.

**Comment by afourney** [comment]
Hey folks. @signalprime ,   sorry for the delay. I've been heads down on a benchmark (GAIA, and hope to be done soon). Enhanced browsing is, I think, planned work for March. 

Having said that, let me raise two points:
1. I've found that Playwright has better support for detecting when pages are loaded, and for interacting with downloads vs. Selenium, and would probably pursue that direction -- but it does not preclude having a common interface that would support Selenium-based approaches.
2. I have refactored websurfer a little, basing it off a new mdconvert package I've been working on that converts various media files to Mardown. Available here:
https://github.com/microsoft/autogen/blob/ct_orchestrator/autogen/mdconvert.py
And Here
https://github.com/microsoft/autogen/blob/ct_orchestrator/autogen/browser_utils.py
